### PR TITLE
Move some rest endpoints to client reader

### PR DESCRIPTION
**Pull request by erikjohnston**

### CODE REVIEW

File: synapse/app/client_reader.py [hunk 1]
```diff
@@ -109,12 +112,12 @@ def _listen_http(self, listener_config):
                     ThreepidRestServlet(self).register(resource)
                     KeyQueryServlet(self).register(resource)
                     KeyChangesServlet(self).register(resource)
+                    VoipRestServlet(self).register(resource)
+                    PushRuleRestServlet(self).register(resource)
```
Comment (richvdh) [inline synapse/app/client_reader.py]: surely you can only safely move GET /pushrules to the worker?

Comment (erikjohnston) [reply synapse/app/client_reader.py]: Oh argh.

Comment (erikjohnston) [reply synapse/app/client_reader.py]: Fixed?

File: synapse/rest/client/v1/push_rule.py [hunk 2]
```diff
@@ -103,6 +107,9 @@ def on_PUT(self, request):
 
     @defer.inlineCallbacks
     def on_DELETE(self, request):
+        if self._is_worker:
+            raise Exception("Cannot handle PUT /push_rules on worker")
```
Comment (richvdh) [inline synapse/rest/client/v1/push_rule.py]: DELETE

**Pull request merged**